MMM-pages.js

```diff
@@ -185,31 +185,51 @@ Module.register('MMM-pages', {
    * Animates the page change from the previous page to the current one. This
    * assumes that there is a discrepancy between the page currently being shown
    * and the page that is meant to be shown.
+   * 
+   * @param {string} [name] - the name of the hiddenPage we want to show. Optional and only used when we want to switch to a hidden page
    */
-  animatePageChange: function () {
+  animatePageChange: function (name) {
     const self = this;
 
-    // Hides all modules not on the current page. This hides any module not
-    // meant to be shown.
-    MM.getModules()
-      .exceptWithClass(this.config.fixed)
-      .exceptWithClass(this.config.modules[this.curPage])
-      .enumerate(module => module.hide(
-        self.config.animationTime / 2,
-        { lockString: self.identifier }
-      ));
-
-    // Shows all modules meant to be on the current page, after a small delay.
-    setTimeout(() => {
+    if (typeof name !== 'undefined') {
+      // Hides all modules not defined in the defined named hidden page.
       MM.getModules()
-        .withClass(self.config.modules[self.curPage])
-        .enumerate((module) => {
-          module.show(
+        .exceptWithClass(this.config.hiddenPages[name])
+        .enumerate(module => module.hide(
+          self.config.animationTime / 2,
+          { lockString: self.identifier }
+        ));
+      // Shows, after a small delay, all modules defined in the selected named hidden page.
+      setTimeout(() => {
+        MM.getModules()
+          .withClass(self.config.hiddenPages[name])
+          .enumerate((module) => {
+            module.show(
+              self.config.animationTime / 2,
+              { lockString: self.identifier }
+            );
+          });
+      }, this.config.animationTime / 2);
+    } else {
+      // Hides all modules not on the current page. This hides any module not
+      // meant to be shown.
+      MM.getModules()
+        .exceptWithClass(this.config.fixed.concat(this.config.modules[this.curPage]))
+        .enumerate(module => module.hide(
+          self.config.animationTime / 2,
+          { lockString: self.identifier }
+        ));
+
+      // Shows all modules meant to be on the current page, after a small delay.
+      setTimeout(() => {
+        MM.getModules()
+          .withClass(self.config.fixed.concat(self.config.modules[self.curPage]))
+          .enumerate(module => module.show(
             self.config.animationTime / 2,
             { lockString: self.identifier }
-          );
-        });
-    }, this.config.animationTime / 2);
+          ));
+      }, this.config.animationTime / 2);
+    }
```

Comment:
Is the only difference between these two if-branches just which set of modules are excluded?

Reply:
Yes, I fixed that in combination with the local variable.